src/UI/Implementation/Component/Button/Toggle.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/* Copyright (c) 2018 Thomas Famula <famula@leifos.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Button;
+
+use ILIAS\UI\Component as C;
+use ILIAS\UI\Implementation\Component\ComponentHelper;
+use ILIAS\UI\Implementation\Component\JavaScriptBindable;
+use ILIAS\UI\Implementation\Component\Triggerer;
+use ILIAS\UI\Implementation\Component\TriggeredSignal;
+use ILIAS\UI\Component\Signal;
+
+class Toggle extends Button implements C\Button\Toggle {
+	use ComponentHelper;
+	use JavaScriptBindable;
+	use Triggerer;
+
+	/**
+	 * @var bool
+	 */
+	protected $is_on;
+
+	/**
+	 * @var string|null
+	 */
+	protected $action_off = null;
+
+	/**
+	 * @var string|null
+	 */
+	protected $action_on = null;
+
+	/**
+	 * @inheritdoc
+	 */
+	public function __construct($label, $action_on, $action_off, $is_on, Signal $click = null)
+	{
+		$this->checkStringOrSignalArg("action", $action_on);
+		$this->checkStringOrSignalArg("action_off", $action_off);
+		$this->checkBoolArg("is_on", $is_on);
+
+		$button_action = (is_null($click))
+			? ""								// no way to resolve conflicting string actions
```

Comment:
Lost of spacing here.

Reply:
Changed that (however this is the usual way I indent =?: expressions). The test if from Richard (a cherry pick we needed).